packages/recs/src/Query.ts

```diff
@@ -441,21 +452,23 @@ export function defineQuery(
             const newMatchingSet = runQuery(fragments, options?.initialSet);
             const updates: (ComponentUpdate & { type: UpdateType })[] = [];
 
-            for (const previouslyMatchingEntity of matching) {
-              // Entity matched before but doesn't match now
+            const currentMatching = matching.value;
+
+            for (const previouslyMatchingEntity of currentMatching) {
               if (!newMatchingSet.has(previouslyMatchingEntity)) {
-                matching.delete(previouslyMatchingEntity);
+                currentMatching.delete(previouslyMatchingEntity);
                 updates.push({
                   entity: previouslyMatchingEntity,
                   type: UpdateType.Exit,
                   component: update.component,
                   value: [undefined, undefined],
                 });
               }
+              matching.next(currentMatching);
```

Comment:
I am wondering if this line is causing the additional render?

Reply:
no, deleting it did not lead to changes.